### DISCUSSION TITLE
HTTP version is not honored when using a proxy

### DISCUSSION THREAD

**Discussion by AlecHaring** [encode/httpx#1960] [Potential Issue] [answered]
Running this code:
```python
import httpx

def http_version_test(url):
    # with proxy
    with httpx.Client(http2=True,
                      proxies={"all://": f"http://127.0.0.1:8080"}) as client:
        res = client.get(url)
        print(f"With proxy: {res.http_version}")

    # without proxy
    with httpx.Client(http2=True) as client:
        res = client.get(url)
        print(f"Without proxy: {res.http_version}")


if __name__ == '__main__':
    http_version_test('https://httpbin.org/get')
```
Will output this:
```
With proxy: HTTP/1.1
Without proxy: HTTP/2
```

It seems like the problem arises in the `HTTPTransport`'s constructor. If `proxy` is not set, it sets `HTTPTransport._pool` to a new `httpcore.ConnectionPool` and passes along the HTTP version arguments. If `proxy` is set, it sets `HTTPTransport._pool` to `httpcore.HTTPProxy` but does not pass along the HTTP version arguments (presumably because the `HTTPProxy` class does not allow you to specify the HTTP version).

**Comment by lovelydinosaur** [comment]
Correct. At this point in time we should simply document that proxy HTTP requests always use HTTP/1.1.

**Reply by AlecHaring** [reply]
Is there a specific reason HTTP2 support was removed from the `HTTPProxy` class in https://github.com/encode/httpcore/commit/f9b93918a54a49a4e917824ad38cf5bd8da21450? Was it just too unstable?

**Reply by lovelydinosaur** [reply]
It's never been supported for proxies. (It's just a fiddly slightly confusing bit of extra work that I've never looked into.)

**Reply by AlecHaring** [reply]
`httpx` has never supported HTTP2 with proxies? If you run the above code using version `0.20.0` you get this:
```
With proxy: HTTP/2
Without proxy: HTTP/2
```

**Reply by lovelydinosaur** [reply]
Okay, that is /not/ what I'd expected, perhaps tunnelled connections *did* work. Worth looking into.

**Comment by p1-ra** [comment]
Hi, 

The following PRs resolved that issue.

- https://github.com/encode/httpx/pull/1647
- https://github.com/encode/httpcore/pull/342

They are waiting feedbacks since May 2021.

Thanks

**Comment by lovelydinosaur** [comment, marked as answer]
Resolved [from 0.2.12 onwards](https://github.com/encode/httpx/blob/master/CHANGELOG.md#0212-5th-january-2022).
